zap/src/main/java/org/parosproxy/paros/network/HttpSender.java

```diff
@@ -615,12 +655,17 @@ private HttpMethod runMethod(HttpMessage msg, boolean isFollowRedirect, HttpMeth
             // cant do this for EntityEnclosingMethod methods - it will fail
             method.setFollowRedirects(isFollowRedirect);
         }
+
+        // needs to verify if the state
+        // should store cookies
+        updateCookieState();
+
         // ZAP: Use custom HttpState if needed
         User forceUser = this.getUser(msg);
         if (forceUser != null) {
             this.executeMethod(method, forceUser.getCorrespondingHttpState());
         } else {
-            this.executeMethod(method, null);
+            this.executeMethod(method, client.getState());
```

Comment:
This should not be necessary (this would have the side effect of enabling cookies for the current request, even if the cookies are actually disabled), the client state is already used internally.